doc/crypto/api/keys/policy.rst

```diff
@@ -302,3 +316,39 @@ The usage flags are encoded in a bitmask, which has the type `psa_key_usage_t`.
     .. admonition:: Implementation note
 
         This is a simple accessor function that is not required to validate its inputs. It can be efficiently implemented as a ``static inline`` function or a function-like-macro.
+
+.. function::  psa_check_key_usage
+
+    .. summary::
+        Queries the capabilities of a PSA key object. 
+
+        .. versionadded:: 1.4
+
+    .. param:: psa_key_id_t key
+        a PSA key identifier.
+
+    .. param:: psa_algorithm_t alg
+        a specific algorithm. 
+
+    .. param:: psa_key_usage_t usage
+         a single PSA_KEY_USAGE_xxx flag. 
+
+    .. return:: psa_status_t
+
+    If the supplied key is a key pair, the function checks the appropriate half of the key pair. For example, if the usage flag was `PSA_KEY_USAGE_SIGN_MESSAGE`, it would check the private key. But if it were `PSA_KEY_USAGE_VERIFY_MESSAGE` it would check the public key. 
```

Comment:
The private key and the public key don't exist as separate objects. A key pair can both sign and verify, unlike a public key which can only verify. There's no need to bring in a “private key” as a separate part of the key pair.